systemtests/auth_test.go

```diff
@@ -106,6 +107,160 @@ func (s *systemtestSuite) TestBuiltInAdminAuthorizationImmortality(c *C) {
+		userToken := loginAs(c, username, username)
+		endpoint := proxy.V1Prefix + "/authorizations"
+
+		// add authorization for built-in ops user using admin token
+		data := `{"PrincipalName":"ops","local":true,"role":"admin","tenantName":""}`
+		opsAuthz := s.addAuthorization(c, data, token)
+
+		opsToken := loginAs(c, "ops", "ops")
+
+		// add, delete and get the authz
+		data = `{"PrincipalName":"` + username + `","local":true,"role":"ops","tenantName":"default"}`
+		// add authorization for `username` using ops token
+		authz := s.addAuthorization(c, data, opsToken)
+		s.deleteAuthorization(c, authz.AuthzUUID, opsToken)
+		resp, body := proxyGet(c, opsToken, endpoint+"/"+authz.AuthzUUID)
+		c.Assert(resp.StatusCode, Equals, http.StatusNotFound)
+		c.Assert(len(body), Equals, 0)
+
+		// delete authz of built-in ops user
+		s.deleteAuthorization(c, opsAuthz.AuthzUUID, token)
+		resp, body = proxyGet(c, token, endpoint+"/"+authz.AuthzUUID)
+		c.Assert(resp.StatusCode, Equals, http.StatusNotFound)
+		c.Assert(len(body), Equals, 0)
+
+		// non-admins cannot access this endpoint
+		resp, _ = proxyDelete(c, userToken, endpoint+"/xxx")
+		c.Assert(resp.StatusCode, Equals, http.StatusForbidden)
+
+		// test 404
+		resp, _ = proxyDelete(c, token, endpoint+"/xxx")
+		c.Assert(resp.StatusCode, Equals, http.StatusNotFound)
+	})
+}
+
+// TestAuthorizationGetEndpoints tests authorization get endpoints (/authorizations, /authorizations/{ID})
+func (s *systemtestSuite) TestAuthorizationGetEndpoints(c *C) {
+	s.addUser(c, username)
+
+	runTest(func(ms *MockServer) {
+		token := adminToken(c)
+		userToken := loginAs(c, username, username)
+		endpoint := proxy.V1Prefix + "/authorizations"
+
+		// built-in admin user is always associated with a authz
+		c.Assert(len(s.getAuthorizations(c, token)) > 1, Equals, true)
+
+		data := `{"PrincipalName":"` + username + `","local":true,"role":"ops","tenantName":"default"}`
+		// add authorization for `username` using admin token
+		authz := s.addAuthorization(c, data, token)
+		c.Assert(authz, DeepEquals, s.getAuthorization(c, authz.AuthzUUID, token))
+
+		// non-admins cannot access this endpoint
+		resp, _ := proxyGet(c, userToken, endpoint)
+		c.Assert(resp.StatusCode, Equals, http.StatusForbidden)
+
+		// add admin authz for `username`
+		data = `{"PrincipalName":"` + username + `","local":true,"role":"admin","tenantName":"default"}`
+		authz = s.addAuthorization(c, data, token)
+		c.Assert(authz.Role, Equals, "admin")
+		// tenantName gets ignored for admin authzs
+		c.Assert(authz.TenantName, Equals, "")
+		c.Assert(authz, DeepEquals, s.getAuthorization(c, authz.AuthzUUID, token))
+
+		// since `username` is an admin after the earlier update, he can access admin APIs now
+		data = `{"PrincipalName":"ops","local":true,"role":"admin","tenantName":""}`
+		s.addAuthorization(c, data, userToken)
+
+		// admin API accessed using user token
+		c.Assert(len(s.getAuthorizations(c, userToken)) >= 4, Equals, true)
```

Comment:
A more clear way to do this would be:

1. store the current number of authorizations
1. call `addAuthorization()`
1. assert that the current number is the original number + 1

Then you can get rid of the magic number and don't need any comments or anything